### DISCUSSION TITLE
Issue with Resend Integration for Notifications (Workflows & useQueryGraphStep)

### DISCUSSION THREAD

**Discussion by amatosc** [medusajs/medusa#10550] [Q&A] [answered]
I have been following the guide to integrate Resend as a module for notifications ([Resend Integration Guide](https://docs.medusajs.com/resources/integrations/guides/resend)), and everything seemed fine at first. However, the problem started when I tried to export information, such as the `display_id` of orders.

The issue is that not only does it throw a type error, but fields like `display_id` and others, such as `shipping_address.*`, are not exported at all.

### Code
```ts
export const sendOrderConfirmationWorkflow = createWorkflow(
  "send-order-confirmation",
  ({ id }: WorkflowInput) => {
    //@ts-ignore
    const { data: orders } = useQueryGraphStep({
      entity: "order",
      fields: [
        "id",
        "email",
        "currency_code",
        "total",
        "items.*",
        "display_id",
        "shipping_address.*",
        "created_at",
        "shipping_methods.*",
      ],
      filters: {
        id,
      },
    });

    // This fails as well despite this -> https://docs.medusajs.com/learn/fundamentals/workflows/multiple-step-usage
    //@ts-ignore
    // const { data: stores } = useQueryGraphStep({
    //   entity: "store",
    //   fields: ["name"],
    // }).config({ name: "fetch-stores" });

    const baseUrl = transform({}, () => process.env.STOREFRONT_URL);

    const notification = sendNotificationStep([
      {
        to: orders[0].email,
        channel: "email",
        template: "order-placed",
        data: {
          order: orders[0],
          baseUrl: baseUrl,
        },
      },
    ]);

    return new WorkflowResponse(notification);
  },
);
```

### Error
```ts
Type '"display_id"' is not assignable to type '"metadata" | "id" | "status" | "created_at" | "updated_at" | "*" | "email" | "currency_code" | "total" | "items.*" | "shipping_address.*" | "shipping_methods.*" | "version" | ... 1189 more ... | "promotion_link.order.raw_original_shipping_tax_total"'.
```

### Data
```json
{
  "id": "noti_01JETG9PK3CVZ83Q9YP9SNZP42",
  "to": "customer@test.com",
  "channel": "email",
  "template": "order-placed",
  "data": {
    "order": {
      "id": "order_01JETG9PD459W33SPSTJ4YXWJZ",
      "email": "customer@test.com",
      "currency_code": "eur",
      "total": 20,
      "items": [
        {...}
      ]
    }
  },
  "provider_id": "resend"
}
```

Additionally, I tried to include more information like urlBase, but it is not included as part of the data.

Steps Taken
- Followed the official guide for integration.
- Attempted to export specific fields (display_id, shipping_address.*).
- Tried to add custom data (urlBase).

Issues Observed
* Type errors when exporting certain fields.
* Specific fields (display_id, shipping_address.*) are not included in the output of orders.
* Custom data like urlBase is also missing in the exported data.

Any guidance on how to resolve this would be greatly appreciated!

**Comment by shahednasser** [comment]
Hi @amatosc you're only getting a type error, right? If you add `// @ts-ignore` before the line with the error, is `display_id` retrieved anyway?

**Reply by amatosc** [reply]
Hi @shahednasser 

With `@ts-ignore`, the type error disappears, but it still doesn’t return `display_id` or any of the other extra fields that weren’t included in the guide, such as `shipping_address` or `created_at`. It also doesn’t include the `baseUrl` field within the `data` field.

![Screenshot 2024-12-11 at 12 58 12](https://github.com/user-attachments/assets/e25e8b90-0436-472c-8985-a05b4749c78b)

**Reply by shahednasser** [reply]
I just tried using your code and I couldn't reproduce the issue. Can you make sure you're running the latest Medusa version by deleting `node_modules` and installing dependencies again? Also make sure to run `npx medusa db:migrate`.

I see you also mentioned that the second `useQueryGraphStep` is failing, can you specify the error you're getting?

**Reply by amatosc** [reply]
I've updated Medusa to the latest version `v2.1.1`, ran the migrations `npx medusa db:migrate`, deleted `node_modules` and `.medusa` directories, and I get the same behavior.

When using `useQueryGraphStep` twice, I get this error:
```logs
error:   Error starting server
error:   Workflow with id "send-order-confirmation" and step definition already exists.
Error: Workflow with id "send-order-confirmation" and step definition already exists.
    at Function.register (/Users/fake/Developer/my-medusa-project/node_modules/.pnpm/@medusajs+orchestration@2.1.1_@mikro-orm+core@5.9.7_@mikro-orm+knex@5.9.7_@mikro-orm+core@5.9_apwe4i44cqwctwbavnzvhaafhe/node_modules/@medusajs/orchestration/src/workflow/workflow-manager.ts:116:15)
    at createWorkflow (/Users/fake/Developer/my-medusa-project/node_modules/.pnpm/@medusajs+workflows-sdk@2.1.1_@mikro-orm+core@5.9.7_@mikro-orm+knex@5.9.7_@mikro-orm+core@5.9_ewtesqnwkqdm2dgqvf3marbcn4/node_modules/@medusajs/workflows-sdk/src/utils/composer/create-workflow.ts:152:21)
    at Object.<anonymous> (/Users/fake/Developer/my-medusa-project/src/workflows/send-order-confirmation.ts:13:60)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module.m._compile (/Users/fake/Developer/my-medusa-project/node_modules/.pnpm/ts-node@10.9.2_@swc+core@1.5.7_@swc+helpers@0.5.15__@types+node@20.17.9_typescript@5.7.2/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/fake/Developer/my-medusa-project/node_modules/.pnpm/ts-node@10.9.2_@swc+core@1.5.7_@swc+helpers@0.5.15__@types+node@20.17.9_typescript@5.7.2/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
Waiting for the debugger to disconnect...
```

**Reply by shahednasser** [reply]
Can you share a repo that I can look into? You shouldn't be running into any of these issues.

**Reply by amatosc** [reply, marked as answer]
I can confirm that everything is working as expected. The previous errors happened because I initially duplicated the workflow file `order-place.ts` for `order-canceled.ts`, but the workflow inside still had the same name, which caused the error. Thank you so much for your help @shahednasser  😊
